src/coreclr/jit/assertionprop.cpp

```diff
@@ -923,7 +923,7 @@ void Compiler::optPrintAssertion(AssertionDsc* curAssertion, AssertionIndex asse
                 break;
 
             case O2K_CONST_DOUBLE:
-                if (*((__int64*)&curAssertion->op2.dconVal) == (__int64)I64(0x8000000000000000))
+                if (*((int64_t*)&curAssertion->op2.dconVal) == (int64_t)I64(0x8000000000000000))
```

Comment:
are we able to transition cases like these to using `std::bitcast` now that we can use the stdlib in more places? It might make them overall more readable

it might even be a case where some internal `IsNegativeZero` helper (in utils with the other floating-point helpers) makes the intent clearer

Reply:
> it might even be a case where some internal IsNegativeZero helper

There is a helper like that in JIT FloatingPointUtils, it is just not used here...